addons/cni-canal/canal.yaml

```diff
@@ -1,4 +1,41 @@
-# Source: https://projectcalico.docs.tigera.io/archive/v3.23/manifests/canal.yaml
+# Source: https://github.com/projectcalico/calico/blob/v3.26.0/manifests/canal.yaml
+---
+# Source: calico/templates/calico-kube-controllers.yaml
+# This manifest creates a Pod Disruption Budget for Controller to allow K8s Cluster Autoscaler to evict
+
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: calico-kube-controllers
+  namespace: kube-system
+  labels:
+    k8s-app: calico-kube-controllers
+spec:
+  maxUnavailable: 1
```

Comment:
I guess we just copy/paste upstream manifests but I'm curious, does this make any sense with
https://github.com/kubermatic/kubeone/blob/c21ab06001fd5d35095eb9182aa76cf850cd1cb2/addons/cni-canal/canal.yaml#L4952-L4953

Reply:
I think the comment above describe the reason for this PDB:
`# This manifest creates a Pod Disruption Budget for Controller to allow K8s Cluster Autoscaler to evict`

Reply:
sure, I was just pondering whether having deployment where it's explicitly stated there can only be 1 replica and then defining PDB with `maxUnavailable=1` doesn't have the exact same behaviour as if no PDB was defined at all.

And I came to the conclusion it probably does but at the same time, there should be no harm in defining PDB.